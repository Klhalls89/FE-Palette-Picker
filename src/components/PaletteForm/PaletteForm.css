.color-form {
  font-family: 'Open Sans', sans-serif;
}

.title-input {
  font-size: 16px;
  padding-left: 5px;
}

.save-pal {
  border: none;
}

.save-pal:hover {
  cursor: pointer;
}
