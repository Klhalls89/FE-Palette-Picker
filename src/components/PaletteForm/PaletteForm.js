import React, { Component } from 'react';

class PaletteForm extends Component {
  constructor() {
    super(); 
    this.state = {
      palette_title: ''
    };
  }

  handleSubmit = e => {
    e.preventDefault();
    const { palette_title } = this.state;
    this.props.handleProject('palette', palette_title);
    this.setState({palette_title: ''});
  };
    
  handleChange = e => {
    const { value } = e.target;
    this.setState({palette_title: value});
  };

  options = () => {
    const {projects} = this.props;
    if (projects.length > 0) {
      return  projects.map(prop => {
        return <option key={Date.now()}>{prop.project_title}</option>;
      });
    }
  }
  
  render() {
    const { palette_title } = this.state;
    return (
      <form className="color-form"
        onSubmit={this.handleSubmit}
        value={palette_title}
      >
        <select>
          {this.options()}
        </select>
        <input 
          className='title-input'
          type="text"
          onChange={this.handleChange}
          value={palette_title}
          placeholder='Enter Palette Name'
        />
        <button className="save-pal">Save Palette</button>
      </form>
    );
  }
}

export default PaletteForm;