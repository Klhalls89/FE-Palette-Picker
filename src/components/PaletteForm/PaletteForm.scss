.title-input {
  font-size: 16px;
  padding-left: 5px;
} 

.save-pal {
  border: none;
  &:hover {
    background-color: #52f71f;
    cursor: pointer;
  }
}