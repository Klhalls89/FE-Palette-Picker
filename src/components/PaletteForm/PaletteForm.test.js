import React from 'react';
import PaletteForm from './PaletteForm.js'
import { shallow } from 'enzyme';
const shortid = require('shortid')


describe('PaletteForm', () => {
  let wrapper;
  let mockProjects;
  let mockHandleProject;

  beforeEach(() => {
    let mockHandleProject = jest.fn()
     mockProjects = [{project_title: 'project one'}]
    wrapper = shallow(<PaletteForm projects={mockProjects} 
                                  handleProject={mockHandleProject}/>)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('options', () => {
    it('should return an array of options', () => {
      wrapper.instance().options()
      expect.arrayContaining(<option></option>)
    });
  });
});
