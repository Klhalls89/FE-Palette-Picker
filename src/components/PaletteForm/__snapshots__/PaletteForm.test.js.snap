// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm should match the snapshot 1`] = `
<form
  className="color-form"
  onMouseEnter={[Function]}
  onSubmit={[Function]}
  value=""
>
  <select>
    <option
      defaultValue={true}
      value=""
    >
      Create Project
    </option>
    <option
      key="Wk9YPOUo5"
      value="project one"
    >
      project one
    </option>
  </select>
  <input
    className="title-input"
    name="palette_title"
    onChange={[Function]}
    placeholder="Enter Palette Name"
    type="text"
    value=""
  />
  <button
    className="save-pal"
  >
    Save Palette
  </button>
</form>
`;
